backend/ibutsu_server/controllers/admin/portal_controller.py

```diff
@@ -0,0 +1,151 @@
+from http import HTTPStatus
+
+import connexion
+from flask import abort
+
+from ibutsu_server.constants import RESPONSE_JSON_REQ
+from ibutsu_server.db.base import session
+from ibutsu_server.db.models import Portal, User
+from ibutsu_server.filters import convert_filter
+from ibutsu_server.util.admin import check_user_is_admin
+from ibutsu_server.util.query import get_offset
+from ibutsu_server.util.uuid import convert_objectid_to_uuid, is_uuid, validate_uuid
+
+
+def admin_add_portal(portal=None, token_info=None, user=None) -> tuple[dict, int]:
+    """Create a portal
+
+    :param body: Portal
+    :type body: dict | bytes
+
+    :rtype: Portal
+    """
+    check_user_is_admin(user)
+    if not connexion.request.is_json:
+        return RESPONSE_JSON_REQ
+    portal = Portal.from_dict(**connexion.request.get_json())
+    # check if portal already exists
+    if portal.id and Portal.query.get(portal.id):
+        return f"Portal id {portal.id} already exist", HTTPStatus.BAD_REQUEST
+    if user := User.query.get(user):
+        portal.owner = user
+    session.add(portal)
+    session.commit()
+    return portal.to_dict(), HTTPStatus.CREATED
+
+
+@validate_uuid
+def admin_get_portal(id_, token_info=None, user=None) -> dict:
+    """Get a single portal by ID
+
+    :param id: ID of test portal
+    :type id: str
+
+    :rtype: Portal
+    """
+    check_user_is_admin(user)
+
+    # get by ID or check if the portal name matches the passed ID
+    if portal := Portal.query.get(id_) or Portal.query.filter(Portal.name == id_).first():
+        return portal.to_dict(with_owner=True)
+    else:
+        abort(HTTPStatus.NOT_FOUND)
```

Comment:
I think maybe we don't need else here

Reply:
correct, I like the explicit behavior of this, but I think it's ultimately subjective. definitely looks cleaner without it.